lib/simplecov/supports/source_file_support.rb

```diff
@@ -0,0 +1,221 @@
+# frozen_string_literal: true
+
+module SimpleCov
+  module Supports
+    module SourceFileSupport
+      ###
+      ## Related to source file lines statistics
+      ###
+
+      # Returns all covered lines as SimpleCov::SourceFile::Line
+      def covered_lines
+        @covered_lines ||= lines.select(&:covered?)
+      end
+
+      # Returns all lines that should have been, but were not covered
+      # as instances of SimpleCov::SourceFile::Line
+      def missed_lines
+        @missed_lines ||= lines.select(&:missed?)
+      end
+
+      # Returns all lines that are not relevant for coverage as
+      # SimpleCov::SourceFile::Line instances
+      def never_lines
+        @never_lines ||= lines.select(&:never?)
+      end
+
+      # Returns all lines that were skipped as SimpleCov::SourceFile::Line instances
+      def skipped_lines
+        @skipped_lines ||= lines.select(&:skipped?)
+      end
+
+      # Returns the number of relevant lines (covered + missed)
+      def lines_of_code
+        covered_lines.size + missed_lines.size
+      end
+
+      # Will go through all source files and mark lines that are wrapped within # :nocov: comment blocks
+      # as skipped.
+      def process_skipped_lines(lines)
+        skipping = false
+        lines.each do |line|
+          if SimpleCov::LinesClassifier.no_cov_line?(line.src)
+            skipping = !skipping
+            line.skipped!
+          elsif skipping
+            line.skipped!
+          end
+        end
+      end
+
+      ## Related to source file branches statistics
+
+      #
+      # Call recursive method that transform our static hash to array of objects
+      # @return [Array]
+      #
+      def build_branches
+        branches_collection(coverage[:branches] || {})
+      end
+
+      #
+      # Recursive method brings all of the branches as array of objects
+      # In logic here we collect only the positive or negative branch,
+      # not the first called branch for it
+      #
+      # @param [Hash] given_branches
+      #
+      # @return [Array]
+      #
+      def branches_collection(given_branches, root_id = nil)
+        @branches_collection ||= []
```

Comment:
we never use `@branches_collection` again, right? Could just be a local variable that we pass along then, right?

Reply:
(renamed though due to method name clash